java/arcs/core/entity/EntityDereferencerFactory.kt

```diff
@@ -32,16 +33,21 @@ class EntityDereferencerFactory(
         if (value == null) return
         when (value) {
             is Reference -> value.dereferencer = create(schema)
-            is RawEntity -> injectDereferencers(schema, value)
+            is RawEntity -> injectDereferencersIntoRawEntity(schema, value)
             is Set<*> -> value.forEach { injectDereferencers(schema, it) }
+            is ReferencableList<*> -> value.value.forEach { injectDereferencers(schema, it) }
         }
     }
 
-    private fun injectDereferencers(schema: Schema, rawEntity: RawEntity) {
+    private fun injectDereferencersIntoRawEntity(schema: Schema, rawEntity: RawEntity) {
         fun injectField(fieldType: FieldType?, fieldValue: Any?) {
```

Comment:
as another readability improvement, what do you think about moving injectField to be a a separate top level private function?

Reply:
Yup, good idea!